### PR TITLE
Add findOrFabricateFlattenedArrayShadowSymbol

### PR DESCRIPTION
**Pull request by a7ehuo**
Add `findOrFabricateFlattenedArrayShadowSymbol`

Signed-off-by: Annabelle Huo <Annabelle.Huo@ibm.com>

### CODE REVIEW

**Comment by 0xdaryl** [comment]
Jenkins build all

**Review by hzongaro — APPROVED** [review]
The changes look good

**Comment by a7ehuo** [comment]
Sorry I should mention, the declaration of `findOrFabricateFlattenedArrayShadowSymbol` in OMR will depend on the final review of https://github.com/eclipse-openj9/openj9/pull/13977. I'll keep `WIP` until the code review completes for  https://github.com/eclipse-openj9/openj9/pull/13977.

File: compiler/compile/OMRSymbolReferenceTable.hpp [hunk 1]
```diff
@@ -672,6 +672,30 @@ class SymbolReferenceTable
...
+    *  \param isFinal
+    *     Specifies whether the field is final.
+    *  \param fieldName
+    *     The name of the field.
+    *  \param fieldSignature
+    *     The signature of the field.
+    *  \return
+    *     Returns an array shadow symbol reference fabricated for the field of a flattened array element.
+    */
+   TR::SymbolReference * findOrFabricateFlattenedArrayShadowSymbol(TR_OpaqueClassBlock *arrayComponentClass, TR::DataType type, uint32_t fieldOffset, bool isVolatile, bool isPrivate, bool isFinal,  const char *fieldName, const char *fieldSignature);
```
Comment (hzongaro) [inline compiler/compile/OMRSymbolReferenceTable.hpp:697]: Sorry for the post-review comment.  Thinking about the name again, I wonder whether a fully descriptive name like `findOrFabricateFlattenedArrayElementFieldShadowSymbol` would be better - or maybe `findOrFabricateFieldOfFlattenedElementShadowSymbol` - as it's not a shadow symbol for the array as a whole that's returned, but a shadow symbol for a field of an array element.

Comment (a7ehuo) [reply compiler/compile/OMRSymbolReferenceTable.hpp]: Addressed in 2ed16c8. The API is also updated to remove `isFinal` and `isVolatile` based on the review comments in https://github.com/eclipse-openj9/openj9/pull/13977.

**Review by jdmpapin — APPROVED** [review]

**Comment by jdmpapin** [comment]
Jenkins build all

**Pull request merged**